spec/firebase_app_distribution_api_client_spec.rb

```diff
@@ -131,6 +130,28 @@
       expect { api_client.upload_binary("app_id", "invalid_binary_path", "android") }
         .to raise_error("#{ErrorMessage.binary_not_found('APK')}: invalid_binary_path")
     end
+
+    it 'retries if a server error occurs' do
+      responses = [
+        [
+          500,
+          {},
+          "Server Error"
+        ],
+        [
+          202,
+          {},
+          {
+            token: "projects/project_id/apps/app_id/releases/-/binaries/binary_hash"
+          }
+        ]
+      ]
+      stubs.post("/app-binary-uploads?app_id=app_id", fake_binary_contents, upload_headers) do |env|
+        responses.shift
+      end
+      api_client.upload_binary("app_id", fake_binary_path, "android")
```

Comment:
Can you also confirm that the `202` response with the `token` is being returned from `upload_binary` call?